### PR TITLE
fix(template): virtual-scroll scrollStrategy dependency error in es2022

### PR DESCRIPTION
**Pull request by samvimes01**
use inject for dependency injection to prevent class instance members undefined errors

When virtual scroll feature used in a project with "target": "es2022" it's broken because class filed uses dependency injection field before initialization
`readonly viewRange = this.scrollStrategy.renderedRange$;`
throws an error and nothing is rendered.

Using inject instead of constructor injection fixes the issue

### CODE REVIEW

File: libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts [hunk 1]
```diff
@@ -142,7 +147,7 @@ export class RxVirtualScrollViewportComponent
    * in sync with the DOM when the next `renderCallback` emitted an event.
    */
   @Output()
-  readonly viewRange = this.scrollStrategy.renderedRange$;
+  readonly viewRange = this.scrollStrategy?.renderedRange$;
```
Comment (edbzn) [inline libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts:150]: I think we can remove the optional chaining here, `scrollStrategy` should always be defined.

File: libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts [hunk 2]
```diff
@@ -151,17 +156,14 @@ export class RxVirtualScrollViewportComponent
    * item actually being visible to the user.
    */
   @Output()
-  readonly scrolledIndexChange = this.scrollStrategy.scrolledIndex$;
+  readonly scrolledIndexChange = this.scrollStrategy?.scrolledIndex$;
```
Comment (edbzn) [inline libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts:159]: Same here.

**Review by edbzn — COMMENTED** [review]
I have added a comment about the optional chaining, could you change it? Thank you for the fix!

File: libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts (hunk 1 above)
Comment (samvimes01) [reply libs/template/experimental/virtual-scrolling/src/lib/virtual-scroll-viewport.component.ts]: done

**Pull request merged**